aiida/schedulers/scheduler.py

```diff
@@ -232,7 +232,13 @@ def _get_run_line(self, codes_info, codes_run_mode):
             else:
                 stderr_str = f'2> {escape_for_bash(code_info.stderr_name)}' if code_info.stderr_name else ''
 
-            output_string = f'{command_to_exec} {stdin_str} {stdout_str} {stderr_str}'
+            command_string = f'{command_to_exec} {stdin_str} {stdout_str} {stderr_str}'
+
+            if code_info.folder_path:
+                run_in_folder = f'cd {escape_for_bash(code_info.folder_path)}'
+                output_string = f'({run_in_folder}; {command_string})'
```

Comment:
Why are there parentheses around the command? Should we instead maybe make it:
```bash
cd {escape_for_bash(code_info.folder_path)} && {command_string}
```
This way if the change directory fails, the command is not run and it fails already there.

Reply:
Running using ``(cd folder; code.x ...)`` will execute ``code.x`` in a sub shell in ``folder``. This way one avoids the ``cd -``. Here it is explained the idea --> [link](https://stackoverflow.com/questions/10566532/how-can-bash-execute-a-command-in-a-different-directory-context).

Reply:
Interesting, I didn't know that bash syntax, thanks for the link. After looking a bit more into it, using [a slightly different syntax may be preferable](https://stackoverflow.com/a/786515/1628893):
```bash
(cd folder && exec executable.x)
```
The `&&` is still probably a good idea to make mistakes in the directory more evident, and the use of `exec` [will replace the shell process with the command and not create a new process](https://www.gnu.org/software/bash/manual/html_node/Bourne-Shell-Builtins.html#Bourne-Shell-Builtins) reducing memory usage.

I think including some link to the bash manual to explain what it does in the docstring would in any case be a good idea.